Tests/FoundationEssentialsTests/AttributedString/AttributedStringConstrainingBehaviorTests.swift

```diff
@@ -566,5 +574,13 @@ class TestAttributedStringConstrainingBehavior: XCTestCase {
         result[result.startIndex ..< result.index(afterCharacter: result.startIndex)] = replacement[replacement.startIndex ..< replacement.index(afterCharacter: str.startIndex)]
         verify(string: result, matches: [("H", nil, nil, "Hello"), ("ello, world", 1, nil, nil)], for: \.testInt, \.testCharacterDependent, \.testString)
     }
-
+    
+    func testInvalidationCharacterInsertionBetweenRuns() {
+        var str = AttributedString("Hello", attributes: .init().testInt(1).testCharacterDependent(2))
+        str += AttributedString("World", attributes: .init().testInt(1).testCharacterDependent(3))
+        
+        // Inserting text between two runs should not invalidate text dependent attributes in either of the surrounding runs
+        str.characters.insert("|", at: str.index(str.startIndex, offsetByCharacters: 5))
+        verify(string: str, matches: [("Hello", 1, 2), ("|", 1, nil), ("World", 1, 3)], for: \.testInt, \.testCharacterDependent)
```

Comment:
Was the previous behavior`[("Hello", 1, nil), ("|", 1, nil), ("World", 1, nil)]`?

Reply:
The previous behavior here would have been `[("Hello", 1, 2), ("|", 1, nil), ("World", 1, 3)]` as well, because we previously ignored invalidation for all insertions. But now that we're handling insertions and invalidating them in other places, I just added this unit test to ensure this doesn't break the invariant that invalidations shouldn't happen when inserting between runs. The test that previously failed without this change is the one above where inserting "A" into "Hello" would result in the surrounding "Hello" still having the text dependent attribute value set.